pkg/channel/consolidated/utils/util.go

```diff
@@ -65,27 +67,32 @@ type KafkaConfig struct {
 	SaramaSettingsYamlString string
 }
 
-func GetKafkaAuthData(ctx context.Context, secretname string, secretNS string) *client.KafkaAuthConfig {
-
-	k8sClient := kubeclient.Get(ctx)
-	secret, err := k8sClient.CoreV1().Secrets(secretNS).Get(ctx, secretname, metav1.GetOptions{})
-
-	if err != nil || secret == nil {
-		logging.FromContext(ctx).Errorf("Referenced Auth Secret not found")
-		return nil
-	}
+func parseTls(secret *corev1.Secret, kafkaAuthConfig *client.KafkaAuthConfig) {
 
-	kafkaAuthConfig := &client.KafkaAuthConfig{}
-	// check for TLS
+	// self-signed CERTs we need CA CERT, USER CERT and KEy
 	if string(secret.Data[TlsCacert]) != "" {
```

Comment:
Curious. Does this allow for client authentication (mTLS) when the CA CERT is not specified (when we use proper CA)? This code seems to ignore the TlsUsercert and TlsUserkey in this case.

Reply:
we dont need them for real CAs.  tested w/ our managed kafka

Reply:
But here, https://github.com/knative-sandbox/eventing-kafka/pull/359/files#diff-0be4a9cfeff3c7254a2121d8dab22f6607fbac51700d2678da775bd841121d33R90  

we're not loading the *client* certificate ( Usercert / Userkey )

Reply:
> we're not loading the client certificate ( Usercert / Userkey )

@maschmid if you need, than in the SECRET, just don't set the `tls.enabled` thing. use all three arguments, like herE:

https://github.com/knative-sandbox/eventing-kafka/blob/main/test/e2e-common.sh#L467-L471

Perhaps the name is not the best ? LOL